src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,32 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
```

Comment:
Consider using `getSessionFactory()` instead of directly accessing `factory` to maintain consistency and clarity, especially if `factory` is a protected field in the superclass `AbstractDao`.